File: packages/baseline-py/src/baseline_py/init/apply_init.py
"""Write the plan, skipping conflicts unless they are forced."""

from baseline_py.init.managed_file import ManagedFile
from baseline_py.init.plan_disposition import PlanDisposition

_WRITTEN = (PlanDisposition.CREATE, PlanDisposition.MERGE)


def apply_init(plan: tuple[ManagedFile, ...], force: bool) -> tuple[ManagedFile, ...]:
    """Write the planned files and return the ones actually written."""
    written: list[ManagedFile] = []
    for managed in plan:
        forced = force and managed.disposition is PlanDisposition.CONFLICT
        if managed.content is None or not (managed.disposition in _WRITTEN or forced):
            continue
        managed.path.parent.mkdir(parents=True, exist_ok=True)
        managed.path.write_text(managed.content, encoding="utf-8")
        written.append(managed)
    return tuple(written)

File: packages/db-quality/src/init/applyInit.ts
import { mkdirSync, writeFileSync } from 'node:fs'
import { dirname, join } from 'node:path'

import type { ManagedFile } from '@/init/ManagedFile.js'

export const applyInit = (root: string, plan: ManagedFile[]): void => {
  for (const file of plan) {
    const writes = file.disposition === 'create' || file.disposition === 'merge'
    if (!writes || file.content === undefined) continue
    mkdirSync(dirname(join(root, file.path)), { recursive: true })
    writeFileSync(join(root, file.path), file.content)
  }
}
